chain/client/src/test_utils/state_witness_distribution_mock.rs

```diff
@@ -0,0 +1,24 @@
+use std::sync::{Arc, Mutex};
+
+use near_async::messaging::CanSend;
+
+use crate::stateless_validation::state_witness_distribution_actions::StateWitnessDistributionActions;
+use crate::stateless_validation::state_witness_distribution_actor::DistributeChunkStateWitnessRequest;
+
+#[derive(Clone)]
+pub struct SynchronousStateWitnessDistributionAdapter {
```

Comment:
Could we possibly get rid of this whole SynchronousXXX thing? It was used for the ShardsManagerActor because there wasn't an easy way to refactor the existing integration tests. This SynchronousXXX pattern is bad and should not be used.

I see that it's being used in the TestEnv. Is this really necessary or could there be some fallback used for testing, such as just not going through the distribution at all and just give the witness to everyone directly?

Not saying this is easy. If it's not, we can leave it.

Reply:
Hmm, this was required to make eh standard set of tests work. I'm not 100% sure if we can remove them but let me give it a try

Reply:
I will eventually get to it... https://github.com/near/nearcore/pull/11152